src/olympia/addons/tasks.py

```diff
@@ -498,3 +506,93 @@ def bump_appver_for_legacy_addons(ids, **kw):
                 addons_to_reindex.append(addon.pk)
     if addons_to_reindex:
         index_addons.delay(addons_to_reindex)
+
+
+def _get_lwt_default_author():
+    try:
+        user = UserProfile.objects.get(
+            email=settings.MIGRATED_LWT_DEFAULT_OWNER_EMAIL)
+    except UserProfile.DoesNotExist:
+        user = UserProfile.objects.create_user(
+            username=None, email=settings.MIGRATED_LWT_DEFAULT_OWNER_EMAIL)
+    return user
+
+
+@transaction.atomic
+def add_static_theme_from_lwt(lwt):
+    upload_zip = build_static_theme_xpi_from_lwt(lwt)
+    # Try to handle LWT with no authors
+    author = (lwt.listed_authors or [_get_lwt_default_author()])[0]
+    # Wrap zip in FileUpload for Addon/Version from_upload to consume.
+    upload = FileUpload.objects.create(
```

Comment:
For consistency we might want to use `FileUpload.from_post()` instead. This would copy the zip to the correct path for `FileUpload`s. Then you wouldn't need to stick anything in `TMP_PATH`.

Reply:
https://github.com/mozilla/addons-server/pull/8240/commits/e778c2609685fed4a91d3cf99f943760a26c4eec